frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -100,25 +102,226 @@ static Parser helpMakeParser(Context* context,
   }
 }
 
+// <7F>HPECHPL
+#define LIBRARY_MAGIC (uint64_t)0x4C5048434550487F
+#define LIBRARY_VERSION_MAJOR 0
+#define LIBRARY_VERSION_MINOR 1
+
+//
+// The library file format (whitespace not significant):
+// <magic number, uint64_t>
+// <library version, major, int>
+// <library version, minor, int>
+// <chpl version, major, int>
+// <chpl version, minor, int>
+// <chpl version, update, int>
+//
+// <user/std module descriptor, std::string>
+//
+// N:<number of BuilderResult entries, uint64_t>
+//   0..N-1: <file path i, std::string><offset i, uint64_t>
+//
+// M:<string cache size, uint64_t>
+//   0..M-1: <id i, int><string length, uint32_t><string, const char*>
```

Comment:
Do we have to store `int` for the IDs or can we just rely on a list ordering? E.g., if the strings are all null terminated we can reconstruct the indices based on their list position.

Reply:
We could do that, and I can add a TODO as future work.